cmd/snap-update-ns/utils.go

```diff
@@ -667,3 +667,92 @@ func createWritableMimic(dir, neededBy string, sec *Secure) ([]*Change, error) {
 	}
 	return changes, nil
 }
+
+// PathIterator traverses through segments (directories and files) of some
+// path. The filesystem is never consulted, traversal is done purely in memory.
+//
+// The iterator is useful in implementing secure traversal of absolute paths
```

Comment:
I think this description is not fully accurate / slightly misleading given the tests below (relative paths).

Reply:
I re-worded everything, please re-read.